app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt

```diff
@@ -39,27 +37,27 @@ class LoginDataSource {
         dbRef.child(DISPATCH_DB_PATH).get().addOnSuccessListener {
             if (!it.exists()) {
                 profilesResultFuture.complete(
-                    Result.Error(IOException("Error retrieving profiles"))
+                    Result.Error(IOException("Error retrieving users"))
                 )
                 return@addOnSuccessListener
             }
-            val profileList = it.value as List<*>
+            val profileList = (it.value as Map<*, *>).values.toList()
```

Comment:
My compiler doesn't seem to accept this line, should be investigated further, but if it works, on the ci I guess it shouldn't be a problem. It says that it.value being an array list cannot be cast to a map.

Reply:
I'm changing the way I'm storing profiles in the db.